File: 75-sort-colors/sort-colors.java
/*
By Rushabh Gala
rushabhvg
GitHub:
https://github.com/rushabhvg
StackOverflow:
https://stackoverflow.com/users/16571212/rushabhvg
LinkedIn:
https://www.linkedin.com/in/rushabhvg/
*/

class Solution {
    public void sortColors(int[] nums) {
        int n = nums.length;
        int pos1 = 0, pos2=n-1;
        for(int i=0; i<=pos2; i++) {
            if(nums[i]==0) {
                // swap(nums[i], nums[pos1++]);
                nums[i] = nums[pos1];
                nums[pos1++] = 0;
            } else if(nums[i]==2) {
                // swap(nums[i--], nums[pos2--]);
                nums[i--] = nums[pos2];
                nums[pos2--] = 2;
            }
        }
    }
}

/*
By Rushabh Gala
rushabhvg
GitHub:
https://github.com/rushabhvg
StackOverflow:
https://stackoverflow.com/users/16571212/rushabhvg
LinkedIn:
https://www.linkedin.com/in/rushabhvg/
*/

File: 75-sort-colors/sort-colors.rs
/*
By Rushabh Gala
rushabhvg
GitHub:
https://github.com/rushabhvg
StackOverflow:
https://stackoverflow.com/users/16571212/rushabhvg
LinkedIn:
https://www.linkedin.com/in/rushabhvg/
*/

impl Solution {
    pub fn sort_colors(nums: &mut Vec<i32>) {
        let mut low = 0;
        let mut high = nums.len() as i32 - 1;
        let mut i = 0;

        while i <= high {
            match nums[i as usize] {
                0 => {
                    nums.swap(i as usize, low as usize);
                    low += 1;
                    i += 1;
                }
                1 => i += 1,
                2 => {
                    nums.swap(i as usize, high as usize);
                    high -= 1;
                }
                _ => panic!("Invalid input"),
            }
        }
    }
}

/*
By Rushabh Gala
rushabhvg
GitHub:
https://github.com/rushabhvg
StackOverflow:
https://stackoverflow.com/users/16571212/rushabhvg
LinkedIn:
https://www.linkedin.com/in/rushabhvg/
*/
